src/index.js

```diff
@@ -1,4 +1,89 @@
-// $(document).ready(function(){
+$(document).ready(function(){
+//selecting html elements for use in color change functions
+document.getElementById("name").onkeydown = function() {nameInputBackgroundChange()};
+document.getElementById("mail").onkeydown = function() {mailInputBackgroundChange()};
+document.getElementById("subject").onkeydown = function() {subjectInputBackgroundChange()};
+document.getElementById("msg").onkeydown = function() {msgInputBackgroundChange()};
+
+//selecting html elements for use in reverting colors back to original color upon focus change
+document.getElementById("name").onchange = function() {nameInputChange()};
+document.getElementById("mail").onchange = function() {mailInputChange()};
+document.getElementById("subject").onchange = function() {subjectInputChange()};
+document.getElementById("msg").onchange = function() {msgInputChange()};
+
+// functions to change input/text area color and background color based on userinput
+function nameInputBackgroundChange() {
+  var nameField = document.getElementById("name");
+  nameField.style.backgroundColor = "white";
+  nameField.style.color = "#4D766E";
+}
+
+function nameInputChange() {
+  var nameField = document.getElementById("name");
+  if (nameField.value === '') {
+    nameField.style.backgroundColor = "#4D766E";
+    nameField.style.color = "white";
+  }
+}
+
+function mailInputBackgroundChange() {
+  var mailField = document.getElementById("mail");
+  mailField.style.backgroundColor = "white";
+  mailField.style.color = "#4D766E";
+}
+
+function mailInputChange() {
+  var mailField = document.getElementById("mail");
+  if (mailField.value === '') {
+    mailField.style.backgroundColor = "#4D766E";
+    mailField.style.color = "white";
+  }
+}
+
+function subjectInputBackgroundChange() {
+  var subjectField = document.getElementById("subject");
+  subjectField.style.backgroundColor = "white";
+  subjectField.style.color = "#4D766E";
+}
+
+function subjectInputChange() {
+  var subjectField = document.getElementById("subject");
+  if (subjectField.value === '') {
+    subjectField.style.backgroundColor = "#4D766E";
+    subjectField.style.color = "white";
+  }
+}
+
+function msgInputBackgroundChange() {
+  var msgField = document.getElementById("msg");
+  msgField.style.backgroundColor = "white";
+  msgField.style.color = "#4D766E";
+}
+
+function msgInputChange() {
+  var msgField = document.getElementById("msg");
+  if (msgField.value === '') {
+    msgField.style.backgroundColor = "#4D766E";
+    msgField.style.color = "white";
+  }
+}
+
+// function to display input values in window alert or alert user of needed inputs
+document.getElementById("contact-send").onclick = function() {contactValueAlert()};
```

Comment:
It would be better for the reader if you define the functions you're using (in this case, `contactValueAlert`), *before* you invoke them.

Reply:
I don't understand what you mean here. 

I'm under the impression that the function is invoked when someone presses the send button. I don't understand how I could define it before it's invoked with that current understanding I have.

Reply:
You'r right, my comment was flawed here. Sorry about that!